### PR TITLE
Nt/swap

### LINKED ISSUE #2722: Swap Feature
Implement swap feature on mobile..
https://ecency.com/market#swap

### PR DESCRIPTION
**Pull request by noumantahir**
### What does this PR?
Allow users to quickly swap funds between HIVE and HBD based on open orders available in market.

### Issue number
fixes #2722 

### Screenshots/Video

demo 1 : master key
https://github.com/ecency/ecency-mobile/assets/6298342/7aac058c-29a8-4db6-8a51-ae3e545f6a3f

demo 2: hive signer
https://github.com/ecency/ecency-mobile/assets/6298342/e54f8e7d-7903-4fc9-9a60-78d1a34874aa

### CODE REVIEW

File: src/providers/ecency/ecency.ts [hunk 1]
```diff
@@ -31,9 +31,9 @@ import {
  * ************************************
  */
 
-export const getCurrencyRate = (currency) =>
+export const getFiatHbdRate = (fiatCode:string) =>
   ecencyApi
-    .get(`/private-api/market-data/${currency}/hbd?fixed=1`)
+    .get(`/private-api/market-data/${fiatCode}/hbd?fixed=1`)
```
Comment (feruzm) [inline src/providers/ecency/ecency.ts:36]: @noumantahir we don't use fixed=1 on website, maybe that's why there is slight difference in prices.

File: src/screens/trade/styles/sawpFeeSection.styles.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { TextStyle, ViewStyle } from 'react-native';
```
Comment (feruzm) [inline src/screens/trade/styles/sawpFeeSection.styles.ts:1]: @noumantahir filename typo

**Review by feruzm — APPROVED** [review]
lgtm

File: src/providers/ecency/ecency.ts (hunk 1 above)
Comment (noumantahir) [reply src/providers/ecency/ecency.ts:36]: seems it has not effect at all... removed it still

Screenshot 2023-07-21 at 5 42 16 PM

Comment (feruzm) [reply src/providers/ecency/ecency.ts:36]: @noumantahir oh, few days ago I have pushed some fixes to vision-api looks like forgot to handle query params. Let me fix that.

Comment (feruzm) [reply src/providers/ecency/ecency.ts:36]: @noumantahir fixed, try https://ecency.com/private-api/market-data/usd/hbd?fixed="1"

**Pull request merged**